### PR TITLE
Alzheimer Detection Using Visual Transfomer Implementation

### PR DESCRIPTION
**Pull request by r-xian**
Project 6 Implementation of Visual Transformer based on the paper “An Image is Worth 16x16 Words:
Transformers for Image Recognition at Scale,” using pytorch to classify images in the ADNI Alzheimers dataset. Hyperparameter tuning with wandb sweeps was used to get the best model of a final accuracy of 60%. Further improvements are noted in the README.

The files provided are:
- `README` - report of the project detailing model, training and results
- `dataset.py` - provides loading of data into pytorch dataloaders with preprocessing of images
- `modules.py` - classes to create the ViT model using pytorch
- `preditc.py` - loads saved model and generates an accuracy using the test set
- `train.py` - trains and saves the ViT model based on the config in the file, logging information to Wandb
- `utils.py` - provides functionality to load, save models, calculate accuracy and creating a new Wandb sweep

Author: Rodger Xiang, 46425067

### CODE REVIEW

**Comment by LinfengLiu98** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme: 
- Excellent and detailed


Commit messages: Excellent

Code:
- Uses Vision transformer
- Well commented
- Good design

Functionality/Performance:
- Did patient-level split
- Model is overfitting
- Validation loss increases during training
- Potential bugs in the codel
- Test acc is 62%


General comments:
- Solves the problem well although performance is poor

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable, missing usage info -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good

**Comment by shakes76** [comment]
Can merge only if the gitignore file is removed. Does not affect grade only the merge

**Comment by r-xian** [comment]
> Can merge only if the gitignore file is removed. Does not affect grade only the merge

hi Shakes, thanks for the feedback, I have removed the gitignore file and should be ready to merge.

**Comment by wangzhaomxy** [comment]
Feedback attempt and no feedback marks lost.
